content/blog/2017-11-28-fragments-in-react-v16.2.md

```diff
@@ -0,0 +1,181 @@
+---
+title: "Fragments in React v16.2"
+author: [clemmy]
+---
+
+Today, we're excited to bring the new `Fragment` export to React, which allows users to render **static** fragments without the unnecessary keying of each element in the fragment!
+
+```jsx
+render() {
+  return (
+    <>
+      <li>Look ma</li>
+      <li>no keys!</li>
+    </>
+  );
+}
+```
+
+### Why Fragments?
+
+When React released version 16, the ability to render an array of elements from a component's `render` method was added. This allowed developers to prevent extraneous markup with `div`s that pollute the HTML. However, it was inconvenient because every element in the array had to be explicitly keyed. Previously, in order to render an array of elements, the following code would be necessary:
+
+```jsx
+render() {
+  return [
+    <span key="first-span">I am</span>,
+    <span key="second-span">static</span>
+  ];
+}
+```
+
+However, the keys are only necessary for React's reconciliation algorithm when child elements are re-arranged and React needs to figure out which item in one render corresponds to in a subsequent render. If the array is static, then keys aren't necessary and only add extra noise to the code. So, let's omit them:
```

Comment:
Let's be consistent about "rearrange" vs "re-arrange". I'd pick "rearrange" everywhere.